src/errors.ts

```diff
@@ -1,6 +1,50 @@
+import { t } from '@lingui/macro'
+
 export class IntegrationError extends Error {
   constructor(message: string) {
     super(message)
     this.name = 'Integration Error'
   }
 }
+
+const DefaultErrorHeader = t`Please refresh the page and try again.`
+const DefaultErrorAction = t`Reload the page`
+
+export abstract class WidgetError extends Error {
+  header: string = DefaultErrorHeader
+  action: string = DefaultErrorAction
+
+  constructor(config: { header?: string; action?: string; message?: string }) {
+    super(config.message)
+    if (config.header) {
```

Comment:
nit: I think this would be clearer with defaulting on one line, in which case you can type it and not call it a default:

    this.header = header ?? DefaultErrorHeader